### DISCUSSION TITLE
Problems installing or update Winget-Autoupdate on some computers

### DISCUSSION THREAD

**Discussion by raulromeral** [Romanitho/Winget-AutoUpdate#643] [Q&A]
Hi,

We are launching this bat file:

@echo off

::::::::::::::::::::::::::::
:: Put WAU Arguments here ::
::::::::::::::::::::::::::::

SET arguments=-UpdatesAtLogon -UpdatesInterval Weekly -InstallUserContext -StartMenuShortcut -DesktopShortcut

::::::::::::::::::::::::::::
:: Run Powershell Script  ::
::::::::::::::::::::::::::::

SET PowershellCmd=Start-Process powershell.exe -Argument '-noprofile -executionpolicy bypass -file "%~dp0Winget-AutoUpdate-Install.ps1" %arguments%
powershell -Command "& {Get-ChildItem -Path '%~dp0' -Recurse | Unblock-File; %PowershellCmd%'}" -Verb RunAs

In some computers it is installed/updated right but on another ones it is not installing it properly, and I couldn't see any error and it does not appear on apps installed. On the computers with previous version (some of them), it is not updating it.

Can you tell me what I can do?

Thanks,

Regards,

**Comment by AndrewDemski-ad-gmail-com** [comment]
Hey @raulromeral 

This might be a tricky investigation of a somewhat complex case.
Let's break it into managed components.

Batch starts PoSh
Posh executes scripts
Scripts install WAU and adds ARP entry

If was to guess, I would look for evidence that PowerShell was executed at all.
A good place to start is checking what was gathered in EventLog.
You may also check if PowerShell is configured te same way on each machine (f.ex. constrained language mode is set to full etc.)

As always it will be something trivial
Good Luck

**Reply by raulromeral** [reply]
All the computers have the same configuration, there is not a common with the computers that are failing. Can you tell me what I can do to debug it?

Thanks,

Regards,

**Reply by AndrewDemski-ad-gmail-com** [reply]
[$VerbosePreference="Continue"](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_preference_variables?view=powershell-5.1)

Good Luck

**Reply by raulromeral** [reply]
I'm not an expert on powershell, can you explain deeply your response?

Thanks!

**Reply by AndrewDemski-ad-gmail-com** [reply]
@raulromeral..

..now or in near future you will need someone who knows PowerShell.
You may need to add it to arsenal of your skills.

I made my baby-steps in PS 3 but definitely started learning PS using books from [Sapien Technologies](https://www.sapien.com/blog/2013/02/06/become-the-master-of-powershell/).

The absolute required minimum would be [TFM 4.0](https://www.sapien.com/blog/2013/10/29/announcing-powershell-4-0-tfm/) - used as introduction to PS (I know, nowadays the min version is 5.1).
That book neatly describes how to think and operate within the context and limitations of the language.

The upgrade from V4 to V5 is focused mostly on improving security and management and V7 is fully capable of running PS scripts written for V5. 

WAU is V5 compatible and V7 can run it without problems.

I wont hide the fact that it will be a path full of adventures, but you must walk it alone.

_I can only show you the door_
_- Morpheus_
